File: LeetCode-Cpp/InvertBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        // Eğer kök düğüm boşsa, hiçbir işlem yapmadan geri dön 
        if (root == NULL) return NULL; 
        // Sol ve sağ çocukları tersine çevir 
        TreeNode* left = invertTree(root->left); 
        TreeNode* right = invertTree(root->right); 
        // Sol ve sağ çocukları değiştir 
        root->left = right; root->right = left; 
        return root;
    }
};

File: LeetCode-Python/InvertBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None: 
            return None 
        # Sol ve sağ çocukları tersine çevir 
        left = self.invertTree(root.left) 
        right = self.invertTree(root.right) 
        # Sol ve sağ çocukları değiştir 
        root.left = right 
        root.right = left 
        return root
